osc-server/src/test/java/org/osc/core/broker/rest/client/k8s/KubernetesStatusApiTest.java

```diff
@@ -0,0 +1,167 @@
+
+public class KubernetesStatusApiTest {
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+
+    @Mock
+    private org.osc.core.broker.rest.client.k8s.KubernetesClient kubernetesClient;
+
+    @Mock
+    private DefaultKubernetesClient fabric8Client;
+
+    @Mock
+    private MixedOperation<ComponentStatus, ComponentStatusList, DoneableComponentStatus, Resource<ComponentStatus,DoneableComponentStatus>> mixedOperationMock;
+
+    private KubernetesStatusApi api;
+
+    @Before
+    public void testInitialize() throws Exception{
+        MockitoAnnotations.initMocks(this);
+        when(this.kubernetesClient.getClient()).thenReturn(this.fabric8Client);
+        this.api = new KubernetesStatusApi(this.kubernetesClient);
+
+        when(this.fabric8Client.componentstatuses()).thenReturn(this.mixedOperationMock);
+    }
+
+    @Test
+    public void testIsServiceReady_WhenK8sClientThrowsKubernetesClientException_ThrowsVmidcException() throws Exception {
+        // Arrange.
+        this.exception.expect(VmidcException.class);
+        when(this.fabric8Client.componentstatuses()).thenThrow(new KubernetesClientException(""));
+
+        // Act.
+        this.api.isServiceReady();
+    }
+
+    @Test
+    public void testIsServiceReady_WhenComponentStatusIsNull_ServiceNotReady() throws Exception {
+        // Arrange.
+        mockComponentStatus(null);
+
+        // Act.
+        boolean isServiceReady = this.api.isServiceReady();
+
+        // Assert.
+        assertFalse(isServiceReady);
+    }
+
+    @Test
+    public void testIsServiceReady_WhenComponentStatusConditionsIsNull_ServiceNotReady() throws Exception {
+        // Arrange.
+        ComponentStatus componentStatus = newComponentStatus(null, null);
+        componentStatus.setConditions(null);
+        mockComponentStatus(componentStatus);
+
+        // Act.
+        boolean isServiceReady = this.api.isServiceReady();
+
+        // Assert.
+        assertFalse(isServiceReady);
+    }
+
+
+    @Test
+    public void testIsServiceReady_WhenComponentStatusHealthyConditionNotFound_ServiceNotReady() throws Exception {
+        // Arrange.
+        ComponentStatus componentStatus = newComponentStatus("other_condition", "True");
+        mockComponentStatus(componentStatus);
+
+        // Act.
+        boolean isServiceReady = this.api.isServiceReady();
+
+        // Assert.
+        assertFalse(isServiceReady);
+    }
+
+
+    @Test
+    public void testIsServiceReady_WhenComponentStatusConditionFalse_ServiceNotReady() throws Exception {
+        // Arrange.
+        ComponentStatus componentStatus = newComponentStatus("Healthy", "False");
+        mockComponentStatus(componentStatus);
+
+        // Act.
+        boolean isServiceReady = this.api.isServiceReady();
+
+        // Assert.
+        assertFalse(isServiceReady);
+    }
+
+    @Test
+    public void testIsServiceReady_WhenComponentStatusConditionFalse_ServiceReady() throws Exception {
```

Comment:
isn't it should be testIsServiceReady_WhenComponentStatusConditionTrue_ServiceReady

Reply:
@karimull i will update the test name above, yes it should be true. For the other comment... we dont need a unit test specific to validate if the component-name is null or wrong because if that is regressed in the code the mocked line when(this.mixedOperationMock.withName("controller-manager")).thenReturn(componentStatusResource); will not work causing many other tests to fail therefore catching the regression. In other words, that mock line is already enforcing that.